pkg/node/node.go

```diff
@@ -112,24 +119,38 @@ func (s *Service) Init() error {
 		var initiators []string
 
 		switch arr.BlockProtocol {
+		case common.NVMETransport:
+			if len(nvmeInitiators) == 0 {
+				return fmt.Errorf("NVMe transport was requested but NVMe initiator is not available")
+			}
+			s.useISCSI = false
+			s.useFC = false
 		case common.ISCSITransport:
 			if len(iscsiInitiators) == 0 {
 				return fmt.Errorf("iSCSI transport was requested but iSCSI initiator is not available")
 			}
+			s.useISCSI = true
 			s.useFC = false
 		case common.FcTransport:
 			if len(fcInitiators) == 0 {
 				return fmt.Errorf("FC transport was requested but FC initiator is not available")
 			}
 			s.useFC = true
+			s.useISCSI = false
 		default:
 			s.useFC = len(fcInitiators) > 0
+			s.useISCSI = len(iscsiInitiators) > 0
 		}
 
 		if s.useFC {
 			initiators = fcInitiators
-		} else {
+			log.Infof("FC Protocol is requested")
+		} else if !s.useFC && s.useISCSI {
```

Comment:
In else condition, s.useFC check is not needed. It is already in else block.

Reply:
Yes removed the !s.useFC condition from else if. Thanks